go/carmen/example_test.go

```diff
@@ -515,3 +519,81 @@ func ExampleDatabase_GetMemoryFootprint() {
 		log.Fatalf("cannot remove dir: %v", err)
 	}
 }
+
+func ExampleHistoricBlockContext_Export() {
+	dir, err := os.MkdirTemp("", "carmen_db_*")
+	if err != nil {
+		log.Fatalf("cannot create temporary directory: %v", err)
+	}
+	db, err := carmen.OpenDatabase(dir, carmen.GetCarmenGoS5WithArchiveConfiguration(), nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := db.AddBlock(uint64(1), func(context carmen.HeadBlockContext) error {
+		if err := context.RunTransaction(func(context carmen.TransactionContext) error {
+			context.CreateAccount(carmen.Address{byte(1)})
+			context.AddBalance(carmen.Address{byte(1)}, carmen.NewAmount(uint64(1)))
+			context.SetState(carmen.Address{byte(1)}, carmen.Key{byte(1)}, carmen.Value{byte(1)})
+			return nil
+		}); err != nil {
+			log.Fatalf("cannot create transaction: %v", err)
+		}
+		return nil
+	}); err != nil {
+		log.Fatalf("cannot add block: %v", err)
+	}
+
+	// flush to sync archive up to latest block
+	if err := db.Flush(); err != nil {
+		log.Fatalf("cannot flush: %v", err)
+	}
+
+	var rootHash carmen.Hash
+	b := bytes.NewBuffer(nil)
+	if err = db.QueryBlock(uint64(1), func(ctxt carmen.HistoricBlockContext) error {
+		rootHash, err = ctxt.Export(context.Background(), b)
+		if err != nil {
+			log.Fatalf("cannot export: %v", err)
+		}
+		return err
+	}); err != nil {
+		log.Fatalf("failed to export block: %v", err)
+	}
+
+	if err := db.Close(); err != nil {
+		log.Fatalf("cannot close db: %v", err)
+	}
+	if err := os.RemoveAll(dir); err != nil {
+		log.Fatalf("cannot remove dir: %v", err)
+	}
+
+	importedDbPath := filepath.Join(os.TempDir(), "imported_carmen")
```

Comment:
This can lead to collisions with already existing copies of previous runs. Better to use
```
os.MkdirTemp("", "carmen_db_*")
```
like in the first line of this test.